tsl/src/process_utility.c

```diff
@@ -278,8 +279,13 @@ tsl_ddl_command_end(EventTriggerData *command)
 					case AT_SetAccessMethod:
 					{
 						Oid relid = AlterTableLookupRelation(stmt, NoLock);
-						bool to_hypercore = (strcmp(cmd->name, TS_HYPERCORE_TAM_NAME) == 0);
-						hypercore_alter_access_method_finish(relid, !to_hypercore);
+						Chunk *chunk = ts_chunk_get_by_relid(relid, false);
```

Comment:
hypercore_alter_access_method_finish() already does this lookup, and I think it should be handled there rather than assuming it has to be a chunk here and then doing the same lookup again. I would recommend changing the same ts_chunk_get_by_relid() to a non-failing lookup in hypercore_alter_access_method_finish()  and then exiting if it is not a chunk.

Reply:
Done.